linux/gnome/src/main/java/com/github/weisj/darkmode/platform/linux/gnome/GnomeThemeMonitorService.java

```diff
@@ -30,17 +30,18 @@
 
     @Override
     public boolean isDarkThemeEnabled() {
-        // TODO: Stop guessing and check the settings when available (like what is mentioned in the GtkVariants class)
         String currentTheme = GnomeNative.getCurrentTheme();
-        return currentTheme.equals(GtkVariants.guessFrom(currentTheme).get(GtkVariants.Variant.Night));
+        if (GnomeSettings.guessLightAndDarkThemes) {
+            return currentTheme.equals(GtkVariants.guessFrom(currentTheme).get(GtkVariants.Variant.Night));
+        } else {
+            return GnomeSettings.darkGtkTheme.getName().equals(currentTheme);
+        }
     }
 
     @Override
     public boolean isHighContrastEnabled() {
-        // TODO: This right now isn't exactly doable with the guessing implementation. It requires a user-accessible
-        // place to
-        // set which theme is their "high contrast theme"
-        return false;
+        String currentTheme = GnomeNative.getCurrentTheme();
```

Comment:
If the guessing mechanism is active this should always return false.

Reply:
Quite right. a0e65a4aa697b013a9ee90a027794ff62d4e10f6